src/status_im/contexts/communities/actions/addresses_for_permissions/events.cljs

```diff
@@ -0,0 +1,58 @@
+(ns status-im.contexts.communities.actions.addresses-for-permissions.events
+  (:require
+    [schema.core :as schema]
+    [taoensso.timbre :as log]
+    [utils.re-frame :as rf]))
+
+(defn get-permissioned-balances
+  [{[event-id _] :original-event} [community-id]]
+  {:fx [[:json-rpc/call
+         [{:method     :wakuext_getPermissionedBalances
+           :params     [{:communityId community-id}]
+           :on-success [:communities/get-permissioned-balances-success community-id]
+           :on-error   (fn [error]
+                         (log/error "failed to get balances"
+                                    {:community-id community-id
+                                     :event        event-id
+                                     :error        error}))}]]]})
+
+(schema/=> get-permissioned-balances
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args [:schema [:? [:catn [:community-id :string]]]]]]
+   [:map {:closed true}
+    [:fx [:vector {:min 1 :max 1} :schema.common/rpc-call]]]])
+
+(rf/reg-event-fx :communities/get-permissioned-balances get-permissioned-balances)
+
+(defn get-permissioned-balances-success
+  [{:keys [db]} [community-id response]]
+  {:db (assoc-in db [:communities/permissioned-balances community-id] response)})
+
+(def ^:private ?account-address keyword?)
+
+(def ^:private ?permissioned-balances-response
+  [:map-of
+   ?account-address
+   [:sequential
+    [:map {:closed true}
+     [:type int?]
+     [:symbol string?]
+     [:decimals int?]
+     [:amount string?]]]])
+
+(schema/=> get-permissioned-balances-success
```

Comment:
@yqrashawn There's a strong pattern in the way we write schemas for event handlers. We could go just a small step further to help alleviate the verbosity for 95% of our use cases. Something to be discussed, but sharing here to give some visibility of the idea.

In the example below, the first argument to `:=>` would be the event args directly, the rest is almost always the same, and when they aren't, the dev would be able to use `schema/=>` instead.

```clojure
(schema/event=> get-permissioned-balances-success
  [:=>
   [:catn
    [:community-id string?]
    [:response ?permissioned-balances-response]]
   [:map {:closed true}
    [:db map?]]])
```

cc @clauxx

Reply:
Would be great to get rid of the boilerplate, I agree. Not as verbose as events, but for quo components we could do the same and just pass a prop map.

Reply:
Yes, we definitely need this.

I'm thinking about something like ring's middleware and `wrap-` functions
Or functions start with `assoc-` or `walk-` to transform schema.
These kinds of functions are easier to document, understand and maintain.

I checked Malli's schema transformation the other day. There are these `get` `assoc` functions implemented for `[LensSchema](https://github.com/metosin/malli/blob/1d6efc23fef01ab333f872b357760de61eac0a21/src/malli/util.cljc#L303)`, but I didn't manage to figure out what's lens schema. Maybe something like [specter](https://github.com/redplanetlabs/specter#readme)?

If that's what we are using. It would be great.

Reply:
> I'm thinking about something like ring's middleware and wrap- functions

@yqrashawn Last week I wrote a malli _transformer_ similar to `malli.util/optional-keys` just to get the feeling of it and to solve one pain point I have in status-mobile. No PR, just experimentation. Transformers are interesting also because they're just functions and we can compose them, which if I understood you correctly is something you would like to have.

OTOH, transformers may be a bazooka to solve a simple problem of just eliminating the most essential boilerplate to define event handler schemas. For that, we just need a vanilla Clojure function and no need to use specific Malli APIs.

> but I didn't manage to figure out what's lens schema.

I only know what lenses are and why they exist in functional languages, but I have zero knowledge about this part of Malli. Looks like something to be explored!

> Maybe something like [specter](https://github.com/redplanetlabs/specter#readme)?

I've never used specter outside playground stuff, but it's surely one of a kind Clojure library. A bit too alien if you ask me.

Reply:
> OTOH, transformers may be a bazooka to solve a simple problem of just eliminating the most essential boilerplate to define event handler schemas. For that, we just need a vanilla Clojure function and no need to use specific Malli APIs.

Totally agree